Library/Homebrew/formula_versions.rb

```diff
@@ -30,7 +30,8 @@ def initialize(formula)
 
   def rev_list(branch)
     repository.cd do
-      Utils.popen_read("git", "rev-list", "--abbrev-commit", "--remove-empty", branch, "--", entry_name) do |io|
+      Utils.popen_read("git", "rev-list", "--abbrev-commit", "--remove-empty", "--first-parent", branch, "--",
```

Comment:
Did you check how much of a difference `--first-parent` on its own makes? It probably doesn't make a difference for homebrew/core, where we don't allow merge commits. (It may well make it faster for coincidental reasons, but I'm not sure this is something we can rely on.)

Reply:
Yes! You can see the results in my benchmark (this is first_parent=false in both the depth=1 and 2 variants). As you note, it doesn't make a big difference. Appears to make things ever so slightly slower, I guess because git is doing a tad more work. When I've inspected the number of revisions it's looked at, the number does drop slightly on a few formula in a private tap repo which doesn't always have a linear history.

I hadn't realized homebrew/core didn't allow merges, so I guess `--first-parent` shouldn't be necessary if just considering homebrew/core. But anywhere merge commits are allowed, I think it is necessary for correctness, if not performance.

Reply:
> I hadn't realized homebrew/core didn't allow merges, so I guess `--first-parent` shouldn't be necessary if just considering homebrew/core. But anywhere merge commits are allowed, I think it is necessary for correctness, if not performance.

Might be worth making this conditional based on the repository?